src/app.ts

```diff
@@ -97,38 +97,78 @@ if (cluster.isMaster) {
 	logger.notice(
 		`open-balena-vpn@${VERSION} process started with pid=${process.pid}`,
 	);
-	spawnChildren(VPN_INSTANCE_COUNT, logger);
-	metricsServer((_req, res) => {
-		for (const clientMetrics of Object.values(workerMetrics)) {
-			metrics.histogram(
-				Metrics.SessionRxBitrate,
-				_.mean(clientMetrics.rxBitrate),
-			);
-			metrics.histogram(
-				Metrics.SessionTxBitrate,
-				_.mean(clientMetrics.txBitrate),
-			);
-		}
-		return new prometheus.AggregatorRegistry()
-			.clusterMetrics()
-			.then((clusterMetrics: string) => {
-				res.set('Content-Type', prometheus.register.contentType);
-				res.write(prometheus.register.metrics());
-				res.write('\n');
-				res.write(clusterMetrics);
-				res.end();
-				workerMetrics = {};
-				metrics.reset(Metrics.SessionRxBitrate);
-				metrics.reset(Metrics.SessionTxBitrate);
+	logger.debug('registering as service instance...');
+	service.wrap(serviceInstance => {
+		logger.info(
+			`registered as service instance with id=${serviceInstance.getId()}`,
+		);
+		logger.info(
+			`spawning ${VPN_INSTANCE_COUNT} worker${
+				VPN_INSTANCE_COUNT > 1 ? 's' : ''
+			}`,
+		);
+		_.times(VPN_INSTANCE_COUNT, i => {
+			const workerId = i + 1;
+			const restartWorker = (code?: number, signal?: string) => {
+				if (signal != null) {
+					logger.crit(`worker-${workerId} killed with signal ${signal}`);
+				}
+				if (code != null) {
+					logger.crit(`worker-${workerId} exited with code ${code}`);
+				}
+				const env = {
+					...process.env,
+					WORKER_ID: workerId,
+					SERVICE_ID: serviceInstance.getId(),
+				};
+				cluster.fork(env).on('exit', restartWorker);
+			};
+			restartWorker();
+		});
+
+		const app = express();
+		app.disable('x-powered-by');
+		app.get('/ping', (_req, res) => res.send('OK'));
```

Comment:
Pretty sure this is changed in master to be an `app.use`

Reply:
Not sure what you mean here? The `vpnWorker` uses `apiServer` (via `app.use`) for its internal API, but this `/ping` relates to the `express` server for exposing prometheus metrics.

```ts
		const app = express();
		app.disable('x-powered-by');
		app.get('/ping', (_req, res) => res.send('OK'));
```

I could abstract these three lines out into a common function, as they are used by both `api.ts` and `app.ts`.

Reply:
I think I was just getting confused with the proxy